sys/include/div.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @brief     Provides integer division functions
+ *
+ * This header provides some integer division functions that can be used
+ * to prevent linking in compiler-generated ones, which are often larger.
+ *
+ * @file
+ * @ingroup   sys_util
+ * @author  Kaspar Schleiser <kaspar@schleiser.de>
+ * @{
+ */
+
+#ifndef DIV_H
+#define DIV_H
+
+#include <assert.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Integer divide val by 15625
+ *
+ * @note This only works for values up to 16383999997,
+ *       otherwise the result is undefined!
+ *
+ * @param[in]   val     dividend
+ * @return      (val / 15625)
+ */
+static inline uint64_t div_u64_by_15625(uint64_t val)
+{
+    assert(val <= 16383999997LLU);
+    return (val * 0x431bde83UL) >> (12 + 32);
+}
+
+/**
+ * @brief Integer divide val by 1000000
+ *
+ * @note This only works for values up to 1048575999808,
+ *       otherwise the result is undefined!
+ *
+ * @param[in]   val     dividend
+ * @return      (val / 1000000)
+ */
+static inline uint64_t div_u64_by_1000000(uint64_t val)
+{
+    assert(val <= 1048575999808LLU);
+    return div_u64_by_15625(val>>6);
+}
+
+/**
+ * @brief Divide val by (15625/512)
+ *
+ * This is used to quantize a 1MHz value to the closest 32768Hz value,
+ * e.g., for timers.
+ *
+ * @param[in]   val     dividend
+ * @return      (val / (15625/512))
+ */
+static inline uint32_t div_u32_by_15625div512(uint32_t val)
+{
+    return ((uint64_t)(val) * 0x431bde83ul) >> (12 + 32 - 9);
+}
+
+/**
+ * @brief Integer divide val by 10
+ *
+ * @param[in]   val     dividend
+ * @return      (val / 10)
+ */
+static inline uint32_t div_u32_by_10(uint32_t n) {
```

Comment:
doxygen will complain about this parameter being `n`, but being documented as `n`